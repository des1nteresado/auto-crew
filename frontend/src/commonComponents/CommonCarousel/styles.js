const styles = (theme) => ({
  carouselContainer: {
    textAlign: 'right',
    marginBottom: 20,
    '& ::-webkit-scrollbar': {
      height: 15,
      [theme.breakpoints.down('xs')]: {
        height: 0,
      },
    },
    '& ::-webkit-scrollbar-track': {
      boxShadow: 'none',
    },
    '& ::-webkit-scrollbar-thumb': {
      backgroundColor: theme.palette.black,
      borderRadius: '100px',
      borderTop: '6px solid white',
      borderBottom: '6px solid white',
    },
    [theme.breakpoints.down('xs')]: {
      marginBottom: 0,
    },
  },
  carouselHeader: {
    display: 'flex',
    justifyContent: 'space-between',
  },
  carousel: {
    display: 'flex',
    width: '100%',
    overflowX: 'scroll',
    '& .MuiCard-root': {
      marginRight: 20,
      marginBottom: 20,
    },
    textAlign: 'left',
  },
  item: {
    marginRight: 24,
    marginBottom: 10,
  },
  title: {
    display: 'block',
    color: theme.palette.black,
    marginBottom: 20,
    [theme.breakpoints.down('xs')]: {
      marginBottom: 10,
    },
  },
  viewAll: {
    width: 'auto',
    marginLeft: 'auto',
    marginBottom: 20,
    padding: 0,
    height: 20,
    [theme.breakpoints.down('xs')]: {
      marginBottom: 10,
    },
  },
});

export default styles;
