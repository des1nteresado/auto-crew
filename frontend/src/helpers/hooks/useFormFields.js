import { useState } from 'react';

export const useFormFields = (initialState) => {
  const [fields, setValue] = useState(initialState);

  return [
    fields,
    (event) => {
      if (event.target.type === 'checkbox') {
        setValue({
          ...fields,
          [event.target.name]: event.target.checked,
        });
      } else {
        setValue({
          ...fields,
          [event.target.name]: event.target.value,
        });
      }
    },
  ];
};
