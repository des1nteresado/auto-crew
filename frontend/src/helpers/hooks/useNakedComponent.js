export const useNakedComponent = (component) => {
  return component?.type?.Naked;
};
