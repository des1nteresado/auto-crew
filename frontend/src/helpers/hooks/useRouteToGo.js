import { useCallback } from 'react';
import { useHistory } from 'react-router-dom';

export const useRouteToGo = (path) => {
  const history = useHistory();

  return useCallback(() => history.push(path), [path, history]);
};
