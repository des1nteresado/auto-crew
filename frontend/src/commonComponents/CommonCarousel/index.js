import React from 'react';
import PropTypes from 'prop-types';
import clsx from 'clsx';
import { withStyles } from '@material-ui/core/styles';
import { Box, Typography } from '@material-ui/core';
import ArrowForwardIosRoundedIcon from '@material-ui/icons/ArrowForwardIosRounded';

import CommonButton from 'commonComponents/CommonButton';

import styles from './styles';

const CommonCarousel = ({
  title,
  classes,
  children,
  withViewAllButton,
  viewAllClickHandler,
  style,
}) => {
  return (
    <Box className={clsx(classes.carouselContainer, style && style)}>
      <Box className={classes.carouselHeader}>
        {title && (
          <Typography className={classes.title} variant="h3">
            {title}
          </Typography>
        )}
        {withViewAllButton && (
          <CommonButton
            endIcon={<ArrowForwardIosRoundedIcon />}
            variant="text"
            label="View All"
            size="small"
            className={classes.viewAll}
            onClick={viewAllClickHandler}
          />
        )}
      </Box>
      <Box className={classes.carousel}>{children}</Box>
    </Box>
  );
};

CommonCarousel.defaultProps = {
  title: '',
  style: '',
  withViewAllButton: false,
  viewAllClickHandler: () => {},
};

CommonCarousel.propTypes = {
  classes: PropTypes.oneOfType([PropTypes.object]).isRequired,
  viewAllClickHandler: PropTypes.func,
  children: PropTypes.node.isRequired,
  withViewAllButton: PropTypes.bool,
  title: PropTypes.string,
  style: PropTypes.string,
};

export default React.memo(withStyles(styles)(CommonCarousel));
