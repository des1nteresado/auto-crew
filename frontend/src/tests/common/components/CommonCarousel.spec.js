import React from 'react';
import { mount } from 'enzyme';

import CommonCarousel from 'common/components/CommonCarousel';
import { useNakedComponent } from 'helpers/hooks/useNakedComponent';

const defaultProps = {
  title: 'title',
  withViewAllButton: true,
  viewAllClickHandler: jest.fn(),
  children: <div>children</div>,
  style: 'style',
};

const setUp = (props) => mount(<CommonCarousel {...defaultProps} {...props} />);

describe('CommonCarousel', () => {
  let component;

  beforeEach(() => {
    component = setUp();
  });

  it('renders CommonCarousel component correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('renders header title when prop is passed correctly', () => {
    const header = component.find('h3');
    expect(header.text()).toBe(defaultProps.title);
  });

  it('renders view all button when withViewAllButton passed correctly', () => {
    const button = component.find('button');
    button.simulate('click');
    expect(defaultProps.viewAllClickHandler).toHaveBeenCalled();
  });

  it('renders commonCarouselContainer elemet with provided styles correctly', () => {
    const div = component.find('div').first();
    expect(div.props().className.includes(defaultProps.style)).toBeTruthy();
  });

  describe('default props', () => {
    const nakedComponent = useNakedComponent(CommonCarousel);

    it('should use default onAutocompleteChange', () => {
      const result = nakedComponent.defaultProps.viewAllClickHandler();
      expect(result).toBe(undefined);
    });
  });
});
