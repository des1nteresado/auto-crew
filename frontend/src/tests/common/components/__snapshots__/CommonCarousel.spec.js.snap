// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonCarousel renders CommonCarousel component correctly 1`] = `
<WithStyles(CommonCarousel)
  style="style"
  title="title"
  viewAllClickHandler={[MockFunction]}
  withViewAllButton={true}
>
  <CommonCarousel
    classes={
      Object {
        "carousel": "CommonCarousel-carousel-3",
        "carouselContainer": "CommonCarousel-carouselContainer-1",
        "carouselHeader": "CommonCarousel-carouselHeader-2",
        "item": "CommonCarousel-item-4",
        "title": "CommonCarousel-title-5",
        "viewAll": "CommonCarousel-viewAll-6",
      }
    }
    style="style"
    title="title"
    viewAllClickHandler={[MockFunction]}
    withViewAllButton={true}
  >
    <Styled(MuiBox)
      className="CommonCarousel-carouselContainer-1 style"
    >
      <div
        className="MuiBox-root MuiBox-root-7 CommonCarousel-carouselContainer-1 style"
      >
        <Styled(MuiBox)
          className="CommonCarousel-carouselHeader-2"
        >
          <div
            className="MuiBox-root MuiBox-root-8 CommonCarousel-carouselHeader-2"
          >
            <WithStyles(ForwardRef(Typography))
              className="CommonCarousel-title-5"
              variant="h3"
            >
              <ForwardRef(Typography)
                className="CommonCarousel-title-5"
                classes={
                  Object {
                    "alignCenter": "MuiTypography-alignCenter",
                    "alignJustify": "MuiTypography-alignJustify",
                    "alignLeft": "MuiTypography-alignLeft",
                    "alignRight": "MuiTypography-alignRight",
                    "body1": "MuiTypography-body1",
                    "body2": "MuiTypography-body2",
                    "button": "MuiTypography-button",
                    "caption": "MuiTypography-caption",
                    "colorError": "MuiTypography-colorError",
                    "colorInherit": "MuiTypography-colorInherit",
                    "colorPrimary": "MuiTypography-colorPrimary",
                    "colorSecondary": "MuiTypography-colorSecondary",
                    "colorTextPrimary": "MuiTypography-colorTextPrimary",
                    "colorTextSecondary": "MuiTypography-colorTextSecondary",
                    "displayBlock": "MuiTypography-displayBlock",
                    "displayInline": "MuiTypography-displayInline",
                    "gutterBottom": "MuiTypography-gutterBottom",
                    "h1": "MuiTypography-h1",
                    "h2": "MuiTypography-h2",
                    "h3": "MuiTypography-h3",
                    "h4": "MuiTypography-h4",
                    "h5": "MuiTypography-h5",
                    "h6": "MuiTypography-h6",
                    "noWrap": "MuiTypography-noWrap",
                    "overline": "MuiTypography-overline",
                    "paragraph": "MuiTypography-paragraph",
                    "root": "MuiTypography-root",
                    "srOnly": "MuiTypography-srOnly",
                    "subtitle1": "MuiTypography-subtitle1",
                    "subtitle2": "MuiTypography-subtitle2",
                  }
                }
                variant="h3"
              >
                <h3
                  className="MuiTypography-root CommonCarousel-title-5 MuiTypography-h3"
                >
                  title
                </h3>
              </ForwardRef(Typography)>
            </WithStyles(ForwardRef(Typography))>
            <CommonButton
              className="CommonCarousel-viewAll-6"
              endIcon={<UNDEFINED />}
              label="View All"
              onClick={[MockFunction]}
              size="small"
              variant="text"
            >
              <WithStyles(ForwardRef(Button))
                className="CommonCarousel-viewAll-6"
                disableFocusRipple={true}
                disableRipple={true}
                disabled={false}
                endIcon={<UNDEFINED />}
                onClick={[MockFunction]}
                size="small"
                variant="text"
              >
                <ForwardRef(Button)
                  className="CommonCarousel-viewAll-6"
                  classes={
                    Object {
                      "colorInherit": "MuiButton-colorInherit",
                      "contained": "MuiButton-contained",
                      "containedPrimary": "MuiButton-containedPrimary",
                      "containedSecondary": "MuiButton-containedSecondary",
                      "containedSizeLarge": "MuiButton-containedSizeLarge",
                      "containedSizeSmall": "MuiButton-containedSizeSmall",
                      "disableElevation": "MuiButton-disableElevation",
                      "disabled": "Mui-disabled",
                      "endIcon": "MuiButton-endIcon",
                      "focusVisible": "Mui-focusVisible",
                      "fullWidth": "MuiButton-fullWidth",
                      "iconSizeLarge": "MuiButton-iconSizeLarge",
                      "iconSizeMedium": "MuiButton-iconSizeMedium",
                      "iconSizeSmall": "MuiButton-iconSizeSmall",
                      "label": "MuiButton-label",
                      "outlined": "MuiButton-outlined",
                      "outlinedPrimary": "MuiButton-outlinedPrimary",
                      "outlinedSecondary": "MuiButton-outlinedSecondary",
                      "outlinedSizeLarge": "MuiButton-outlinedSizeLarge",
                      "outlinedSizeSmall": "MuiButton-outlinedSizeSmall",
                      "root": "MuiButton-root",
                      "sizeLarge": "MuiButton-sizeLarge",
                      "sizeSmall": "MuiButton-sizeSmall",
                      "startIcon": "MuiButton-startIcon",
                      "text": "MuiButton-text",
                      "textPrimary": "MuiButton-textPrimary",
                      "textSecondary": "MuiButton-textSecondary",
                      "textSizeLarge": "MuiButton-textSizeLarge",
                      "textSizeSmall": "MuiButton-textSizeSmall",
                    }
                  }
                  disableFocusRipple={true}
                  disableRipple={true}
                  disabled={false}
                  endIcon={<UNDEFINED />}
                  onClick={[MockFunction]}
                  size="small"
                  variant="text"
                >
                  <WithStyles(ForwardRef(ButtonBase))
                    className="MuiButton-root MuiButton-text CommonCarousel-viewAll-6 MuiButton-textSizeSmall MuiButton-sizeSmall"
                    component="button"
                    disableRipple={true}
                    disabled={false}
                    focusRipple={false}
                    focusVisibleClassName="Mui-focusVisible"
                    onClick={[MockFunction]}
                    type="button"
                  >
                    <ForwardRef(ButtonBase)
                      className="MuiButton-root MuiButton-text CommonCarousel-viewAll-6 MuiButton-textSizeSmall MuiButton-sizeSmall"
                      classes={
                        Object {
                          "disabled": "Mui-disabled",
                          "focusVisible": "Mui-focusVisible",
                          "root": "MuiButtonBase-root",
                        }
                      }
                      component="button"
                      disableRipple={true}
                      disabled={false}
                      focusRipple={false}
                      focusVisibleClassName="Mui-focusVisible"
                      onClick={[MockFunction]}
                      type="button"
                    >
                      <button
                        className="MuiButtonBase-root MuiButton-root MuiButton-text CommonCarousel-viewAll-6 MuiButton-textSizeSmall MuiButton-sizeSmall"
                        disabled={false}
                        onBlur={[Function]}
                        onClick={[MockFunction]}
                        onDragLeave={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        tabIndex={0}
                        type="button"
                      >
                        <span
                          className="MuiButton-label"
                        >
                          View All
                          <span
                            className="MuiButton-endIcon MuiButton-iconSizeSmall"
                          >
                            <ForwardRef>
                              <WithStyles(ForwardRef(SvgIcon))>
                                <ForwardRef(SvgIcon)
                                  classes={
                                    Object {
                                      "colorAction": "MuiSvgIcon-colorAction",
                                      "colorDisabled": "MuiSvgIcon-colorDisabled",
                                      "colorError": "MuiSvgIcon-colorError",
                                      "colorPrimary": "MuiSvgIcon-colorPrimary",
                                      "colorSecondary": "MuiSvgIcon-colorSecondary",
                                      "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                      "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                      "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                      "root": "MuiSvgIcon-root",
                                    }
                                  }
                                >
                                  <svg
                                    aria-hidden={true}
                                    className="MuiSvgIcon-root"
                                    focusable="false"
                                    viewBox="0 0 24 24"
                                  >
                                    <path
                                      d="M7.38 21.01c.49.49 1.28.49 1.77 0l8.31-8.31c.39-.39.39-1.02 0-1.41L9.15 2.98c-.49-.49-1.28-.49-1.77 0s-.49 1.28 0 1.77L14.62 12l-7.25 7.25c-.48.48-.48 1.28.01 1.76z"
                                    />
                                  </svg>
                                </ForwardRef(SvgIcon)>
                              </WithStyles(ForwardRef(SvgIcon))>
                            </ForwardRef>
                          </span>
                        </span>
                      </button>
                    </ForwardRef(ButtonBase)>
                  </WithStyles(ForwardRef(ButtonBase))>
                </ForwardRef(Button)>
              </WithStyles(ForwardRef(Button))>
            </CommonButton>
          </div>
        </Styled(MuiBox)>
        <Styled(MuiBox)
          className="CommonCarousel-carousel-3"
        >
          <div
            className="MuiBox-root MuiBox-root-9 CommonCarousel-carousel-3"
          >
            <div>
              children
            </div>
          </div>
        </Styled(MuiBox)>
      </div>
    </Styled(MuiBox)>
  </CommonCarousel>
</WithStyles(CommonCarousel)>
`;
